core/src/sumeragi/network_topology.rs

```diff
@@ -181,7 +182,23 @@ impl Topology {
         view_change_index: u64,
         new_peers: UniqueVec<PeerId>,
     ) -> Self {
-        let mut topology = Topology::new(block.commit_topology().clone());
+        let created_with_vrf_state = block.header().vrf_state.clone();
+        let mut rng = StdRng::seed_from_u64(u64::from_le_bytes(
+            HashOf::new(&created_with_vrf_state).as_ref()[0..8]
+                .try_into()
+                .expect("Cannot fail"),
+        ));
+
+        let mut topology = {
+            let mut shuffle_peers: Vec<PeerId> = block.commit_topology().clone().into();
+            shuffle_peers.shuffle(&mut rng);
+            let mut ordered_peers = UniqueVec::new();
+            ordered_peers.extend(shuffle_peers);
+            Topology {
+                ordered_peers,
+                created_with_vrf_state,
+            }
+        };
```

Comment:
I noticed that we still just rotate topology on view change witch create predictable peer order, should we do something about that?

Reply:
We can't do anything about that. Any order change will be predictable since all peers must calculate it separately. This is only true for the view change though.